promotion/datastore.go

```diff
@@ -153,6 +157,68 @@ func NewPostgres(databaseURL string, performMigration bool) (*Postgres, error) {
 	return pg, nil
 }
 
+// CreateOrder given the promotion type, initial number of grants and the desired value of those grants
+func (pg *Postgres) CreateOrder(totalPrice decimal.Decimal, merchantID string, status string, orderItems []OrderItem) (*Order, error) {
+	tx := pg.DB.MustBegin()
+
+	var orderID uuid.UUID
+	err := tx.Get(&orderID,
+		`WITH new_order AS (
+			INSERT INTO orders (total_price, merchant_id, status)
+			VALUES ($1, $2, $3)
+			RETURNING id
+		) SELECT id from new_order;`,
+		totalPrice, merchantID, status)
+
+	if err != nil {
+		return nil, err
```

Comment:
are rollbacks being handled?

Reply:
Yep! With the sqlx interface using  `tx, err := db.Begin()` and `tx.Commit()` this will handle rollbacks

Reply:
https://github.com/brave-intl/bat-go/pull/226/commits/6f9661323535bf6dbb30508c797d2ffca5f639b9